src/core/provider.cc

```diff
@@ -743,15 +746,16 @@ InferResponseProvider::AllocateOutputBuffer(
           "Classification result is requested for output '" + name + "'" +
               " while its output buffer size is 0");
     }
-    if (preferred_memory_type == TRTSERVER_MEMORY_CPU) {
-      loutput->cls_count_ = pr->second.cls().count();
-      char* buffer = new char[content_byte_size];
-      *content = static_cast<void*>(buffer);
-      loutput->ptr_ = static_cast<void*>(buffer);
-      loutput->buffer_.reset(buffer);
-    } else {
-      return Status::Success;
+    if (preferred_memory_type != TRTSERVER_MEMORY_CPU) {
```

Comment:
If the output is class type instead of Raw and memory is GPU then we set the actual memory type to CPU and device id to 0 here. It needs to be done before the alloc_fn is called

Reply:
actual* are outputs from the alloc_fn so we don't need to set them before. And the preferred doesn't matter since we are requesting 0 size.

Reply:
Understood